### DISCUSSION TITLE
How to send embed with button

### DISCUSSION THREAD

**Discussion by fakezozi** [Rapptz/discord.py#8335] [Q&A]
I need to send the embed with button but idk how to do this
 ```python
await ctx.send(embed=emb, component=[Button(style=discord.ButtonStyle.red, label = "label1"]

interaction = await bot.wait_for('interaction', check=lambda i: i.label== label1')
await interaction.response.edit_message(content='Button clicked!', view=None)```

**Comment by LostLuma** [comment]
Hi there, this code seems to mix discord.py code with code from unsupported third-party libraries.

To send buttons alongside other message content you will want to create a view instead and create the button inside of it:

```py
class MyView(discord.ui.View):
    @discord.ui.button(label='label1', style=discord.ButtonStyle.red)
    async def on_button_click(self, interaction: discord.Interaction, button: discord.ui.Button) -> None:
        await interaction.response.edit_message(content='Button clicked!', view=None)
```

You can then send this view alongside any message if you'd like to:

```py
embed = discord.Embed(description='Hello, World!')
await channel.send(embed=embed, view=MyView())
```

Since you're using `wait_for` to wait for the user to click on the button you probably want to take a look at the [confirm example](https://github.com/Rapptz/discord.py/blob/master/examples/views/confirm.py) as it achieves this with views, but can work if multiple messages have the same buttons on them simultaneously without any extra effort on your part.

There is also a new section in the docs on the [Bot UI Kit](https://discordpy.readthedocs.io/en/latest/interactions/api.html#bot-ui-kit) and quite a few [other view examples](https://github.com/Rapptz/discord.py/tree/master/examples/views) that might be helpful to you.
